### PR TITLE
sys-block/rts_pstor: version bump to 1.10_p20160103

### PR DESCRIPTION
**Pull request by madmartin**
change upstream source to github
https://github.com/gexplorer/RTS5209-linux-driver
change to EAPI7

Closes: https://bugs.gentoo.org/662000
Package-Manager: Portage-2.3.51, Repoman-2.3.12
Signed-off-by: Martin Dummer <martin.dummer@gmx.net>

### CODE REVIEW

**Comment by madmartin** [comment]
@pacho2 you put sys-block/rts_pstor to maintainer-needed
I offer to maintain it as proxy maintainer, because I need this driver for my notebook.

File: sys-block/rts_pstor/Manifest [hunk 1]
```diff
@@ -1 +1,2 @@
 DIST rts_pstor-1.10.tar.bz2 79549 BLAKE2B 6ed1a31c46920f30b16c3cd4bda9978bffe40c2f88fdd2d63ef128d3a4ce22a6850846e4f5f1bf55200ed03a227a61bfcaefec4abdf598a43c8a7d8ca2991280 SHA512 26d29ccda55672adab8b96a8318e240d0c9d50ebf614853284cf218058c5788cc83b818ce118ac313c0ce6a67aef8ad13d93ed26212db7e05db51b8a7860b48d
```
Comment (mgorny) [inline sys-block/rts_pstor/Manifest]: Add version to the commit message, on the summary line (i.e. `bump to XXX`).

File: sys-block/rts_pstor/metadata.xml [hunk 2]
```diff
@@ -1,7 +1,15 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <!DOCTYPE pkgmetadata SYSTEM "http://www.gentoo.org/dtd/metadata.dtd">
 <pkgmetadata>
-	<!--maintainer-needed-->
+	<maintainer type="person">
+		<email>martin.dummer@gmx.net</email>
+		<name>Martin Dummer</name>
+		<description>Proxied maintainer</description>
```
Comment (mgorny) [inline sys-block/rts_pstor/metadata.xml]: Don't state the obvious.

File: sys-block/rts_pstor/rts_pstor-1.10_p20160103.ebuild [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+inherit linux-mod
+
+GIT_COMMIT="c8313abffe083ac63bf76d2cc90d3edf5b2d1188"
+
+DESCRIPTION="PCIE RTS5209 card reader driver for Linux"
+HOMEPAGE="https://github.com/gexplorer/RTS5209-linux-driver"
+SRC_URI="https://github.com/gexplorer/RTS5209-linux-driver/archive/${GIT_COMMIT}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="GPL-2"
```
Comment (mgorny) [inline sys-block/rts_pstor/rts_pstor-1.10_p20160103.ebuild]: Please verify it's not GPL-2+.

File: sys-block/rts_pstor/rts_pstor-1.10_p20160103.ebuild [hunk 4]
```diff
@@ -0,0 +1,39 @@
+# Copyright 2018 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+inherit linux-mod
+
+GIT_COMMIT="c8313abffe083ac63bf76d2cc90d3edf5b2d1188"
+
+DESCRIPTION="PCIE RTS5209 card reader driver for Linux"
```
Comment (mgorny) [inline sys-block/rts_pstor/rts_pstor-1.10_p20160103.ebuild]: Isn't 'PCI-E' the correct spelling?

File: sys-block/rts_pstor/rts_pstor-1.10_p20160103.ebuild [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+HOMEPAGE="https://github.com/gexplorer/RTS5209-linux-driver"
+SRC_URI="https://github.com/gexplorer/RTS5209-linux-driver/archive/${GIT_COMMIT}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="GPL-2"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE=""
+
+DEPEND=""
```
Comment (mgorny) [inline sys-block/rts_pstor/rts_pstor-1.10_p20160103.ebuild]: You can remove them if they're really empty.

File: sys-block/rts_pstor/rts_pstor-1.10_p20160103.ebuild [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+RDEPEND="${DEPEND}"
+
+S="${WORKDIR}/RTS5209-linux-driver-${GIT_COMMIT}"
+
+MODULE_NAMES="rts_pstor(misc/drivers/scsi)"
+BUILD_TARGETS="default"
+CONFIG_CHECK="~!MISC_RTSX_PCI"
+ERROR_MISC_RTSX_PCI="CONFIG_MISC_RTSX_PCI: The in-kernel driver rtsx_pci is configured, which may have the same functionality than this driver. Make sure that your kernel loads only one of both drivers - use a module blacklisting."
```
Comment (mgorny) [inline sys-block/rts_pstor/rts_pstor-1.10_p20160103.ebuild]: Maybe install a modprobe.d file for it?

File: sys-block/rts_pstor/rts_pstor-1.10_p20160103.ebuild [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+
+src_prepare() {
+	default
+
+	sed -e 's#make -C \/lib\/modules\/\$(shell uname -r)\/build\/#+make -C \$(KERNELDIR)#' -i Makefile || die "sed failed!"
```
Comment (mgorny) [inline sys-block/rts_pstor/rts_pstor-1.10_p20160103.ebuild]: This really shouts for a patch. Also, do you really need to pass it explicitly? I dare say other modules work without such hacks.

Comment (mgorny) [reply sys-block/rts_pstor/rts_pstor-1.10_p20160103.ebuild]: I mean other kernel modules than this one. They get this right without having to hack it over in every  ebuild.

File: sys-block/rts_pstor/rts_pstor-1.10_p20160103.ebuild (hunk 3 above)
Comment (madmartin) [reply sys-block/rts_pstor/rts_pstor-1.10_p20160103.ebuild]: You're right, code is GPL-2+.

File: sys-block/rts_pstor/rts_pstor-1.10_p20160103.ebuild (hunk 4 above)
Comment (madmartin) [reply sys-block/rts_pstor/rts_pstor-1.10_p20160103.ebuild]: Wikipedia says PCI-E or PCIe

File: sys-block/rts_pstor/rts_pstor-1.10_p20160103.ebuild (hunk 5 above)
Comment (madmartin) [reply sys-block/rts_pstor/rts_pstor-1.10_p20160103.ebuild]: Ok.

File: sys-block/rts_pstor/rts_pstor-1.10_p20160103.ebuild (hunk 6 above)
Comment (madmartin) [reply sys-block/rts_pstor/rts_pstor-1.10_p20160103.ebuild]: Yes, can do that, but isn't that a little bit too patronising?

File: sys-block/rts_pstor/rts_pstor-1.10_p20160103.ebuild (hunk 7 above)
Comment (madmartin) [reply sys-block/rts_pstor/rts_pstor-1.10_p20160103.ebuild]: I tried it out - patch file is ~700 byte in size. sed instruction + BUILD_PARAMS setting together ~200 byte. You choose, please.

File: sys-block/rts_pstor/rts_pstor-1.10_p20160103.ebuild (hunk 6 above)
Comment (mgorny) [reply sys-block/rts_pstor/rts_pstor-1.10_p20160103.ebuild]: I think that's what nvidia-drivers are doing. Principle of least surprise, I guess.

File: sys-block/rts_pstor/rts_pstor-1.10_p20160103.ebuild (hunk 7 above)
Comment (mgorny) [reply sys-block/rts_pstor/rts_pstor-1.10_p20160103.ebuild]: You can create smaller patches by controlling context. Sed instruction has the main problem that it returns successfully even if it doesn't substitute anything.

File: sys-block/rts_pstor/rts_pstor-1.10_p20160103.ebuild (hunk 6 above)
Comment (madmartin) [reply sys-block/rts_pstor/rts_pstor-1.10_p20160103.ebuild]: automatic blacklist for in-tree module is now in ebuild

File: sys-block/rts_pstor/rts_pstor-1.10_p20160103.ebuild (hunk 7 above)
Comment (madmartin) [reply sys-block/rts_pstor/rts_pstor-1.10_p20160103.ebuild]: converted to minimum-style patch

**Comment by madmartin** [comment]
@mgorny final commit for now. please review changes,

File: sys-block/rts_pstor/metadata.xml (hunk 2 above)
Comment (mgorny) [reply sys-block/rts_pstor/metadata.xml]: You still state the obvious. Remove the useless description.

Comment (madmartin) [reply sys-block/rts_pstor/metadata.xml]: yeah. removed.

**Pull request closed without merging**